driver-core/src/main/com/mongodb/internal/operation/QueryBatchCursor.java

```diff
@@ -320,83 +322,311 @@ private BsonDocument asGetMoreCommandDocument() {
         return document;
     }
 
-    private void initFromQueryResult(final QueryResult<T> queryResult) {
-        serverCursor = queryResult.getCursor();
+    @Nullable
+    private ServerCursor initFromQueryResult(final QueryResult<T> queryResult) {
         nextBatch = queryResult.getResults().isEmpty() ? null : queryResult.getResults();
         count += queryResult.getResults().size();
+        return queryResult.getCursor();
     }
 
-    private void initFromCommandResult(final BsonDocument getMoreCommandResultDocument) {
-        QueryResult<T> queryResult = getMoreCursorDocumentToQueryResult(getMoreCommandResultDocument.getDocument(CURSOR),
-                                                                        connectionSource.getServerDescription().getAddress());
+    @Nullable
+    private ServerCursor initFromCommandResult(final BsonDocument getMoreCommandResultDocument) {
+        QueryResult<T> queryResult = getMoreCursorDocumentToQueryResult(getMoreCommandResultDocument.getDocument(CURSOR), serverAddress);
         postBatchResumeToken = getPostBatchResumeTokenFromResponse(getMoreCommandResultDocument);
         operationTime = getMoreCommandResultDocument.getTimestamp(OPERATION_TIME, null);
-        initFromQueryResult(queryResult);
+        return initFromQueryResult(queryResult);
     }
 
     private boolean limitReached() {
         return Math.abs(limit) != 0 && count >= Math.abs(limit);
     }
 
-    private void killCursor() {
-        if (serverCursor != null && killCursorOnClose) {
+    private BsonDocument getPostBatchResumeTokenFromResponse(final BsonDocument result) {
```

Comment:
This is not a new method. Its position has changed as a result of removing multiple methods (their logic went to `ResourceManger`), and the diff tool decided it is new.